Samples-Typescript/DashboardLayout/dashboardLayout.ts

```diff
@@ -0,0 +1,155 @@
+import { DashboardLayoutChange, DashboardLayoutChangedEvent, DashboardObject, TableauEvent } from '@tableau/extensions-api-types';
+
+// Wrap everything in an anonymous function to avoid polluting the global namespace
+(async () => {
+  class DashboardLayout {
+    public dashboardObjects: DashboardObject[];
+    private self: DashboardLayout;
+
+    // Avoid globals.
+    constructor(private _$: JQueryStatic) {}
+
+    /**
+     * Initializes the extension
+     */
+    public async initialize() {
+      console.log('Waiting for DOM ready');
+      await this._$.ready;
+      console.log('Initializing extension API');
+      await tableau.extensions.initializeAsync();
+
+      const dashboard = tableau.extensions.dashboardContent.dashboard;
+      this.dashboardObjects = dashboard.objects;
+      console.log(this.dashboardObjects);
+
+      // enabling dashboard event button
+      this._$('#dashboard-event-btn').prop('disabled', false);
+      this._$('#dashboard-event-btn').click(this.onEventButtonClick.bind(this));
+    }
+
+    // When changes are made to the dashboard we get all the details for each of the
+    // dashboard objects that were changed and compare it with their previous values.
+    // The dashboardLayoutChangeDetails property is a map of dashboard obj3ct ids to
+    // an array of dashboard layout changes.
+    // Dashboard layout change events are invoked when dashboard objects are resized, repositioned, added, removed,
+    // hidden, shown, renamed, made floating, and made tiled. Extension reloads when worksheets are added / removed.
```

Comment:
This comment can get out of date. It looks like an exhaustive list, but new functionality could also cause layout changed events to fire, which would be documented in the event documentation. 
Can you change the list of events to 'possible' triggers, and maybe link or mention the documentation for dashboardlayoutchange?

We will not remember to update this comment so it should be more vague.